### PR TITLE
Implementation of comparision of different compression algos

### PR DESCRIPTION
**Pull request by ASBJ021**
Please check implementation of different compression algorithms Eg: FOR, delta

### CODE REVIEW

File: compare_compression_algo.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                                    original_dataset=original_dataset_,
+                                    encoding_scheme= comp_algo,
+                                    enlarge_factor=10)
+    # num_images = len(original_dataset_)
+        num_images = 100
+        print(f"#### Compressing  using {comp_algo} and storing {num_images} images #### ")
+        comp_start_time = time.time()
+        for idx in range(num_images):
+            tensor_storage.add()
+        comp_end_time = time.time()
```
Comment (suman1209) [inline compare_compression_algo.py:37]: isnt this the compression time for 100 images, finally we need to divide by the number of images i think to get the average compression time

File: compare_compression_algo.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        total_comp_time = comp_end_time - comp_start_time
+        print(f'Compression Time of {comp_algo}: {total_comp_time} ')
+
+        decomp_start_time = time.time()
+        tensor_storage.get_image(indx)
```
Comment (suman1209) [inline compare_compression_algo.py:43]: shouldnt we take the average of the decompression time for 100 images here?

**Review by suman1209 — CHANGES_REQUESTED** [review]

File: compare_compression_algo.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        total_decomp_time = decomp_end_time - decomp_start_time
+        print(f'Decompression Time of {comp_algo}: {total_decomp_time} ')
+
+        comp_size = tensor_storage.get_size()
+        orig_size = original_dataset_.get_storage_size()
+
+        print(f"total size in kB of tensor storage: {comp_size}")
+        print(f"total size in kB of original dataset: {orig_size}")
+
+        comp_ratio = orig_size/comp_size
```
Comment (suman1209) [inline compare_compression_algo.py:55]: also all the values can you correct them to the nearest 3sf, so that the final table alignment becomes right

**Comment by suman1209** [comment]
https://pypi.org/project/prettytable/
@ASBJ021 can you have a look at this. You can get a cleaner table of results with this method

**Comment by ASBJ021** [comment]
Sure

**Comment by suman1209** [comment]
@ASBJ021 please update the requirements file

**Review by suman1209 — APPROVED** [review]

**Pull request merged**